### PR TITLE
Update set to not deep merge and add swap

### PR DESCRIPTION
**Pull request by KidkArolis**
First PR part of tiny-atom 4.0.0

cc @KwanMan

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -138,7 +152,7 @@ atom.dispatch('fetchMovies')
 atom.dispatch('increment', 5)
 ```
 
-### `atom.observe(cb)`
+### `atom.observe(cb`
```
Comment (KwanMan) [inline README.md]: hmmm

File: README.md [hunk 2]
```diff
@@ -129,6 +126,23 @@ atom.get()
 atom.get().feed.items
 ```
 
+### `atom.set(update)`
+
+Update current state by merging the update shallowly.
+
+```js
+atom.set({ user })
+atom.set({ entities: { ...entities, posts } })
```
Comment (KwanMan) [inline README.md]: a bit confusing at first glance here with the double `entities`.

```js
atom.set({ entities: { ...prevState.entities, posts } })
```
?

File: README.md [hunk 3]
```diff
@@ -129,6 +126,23 @@ atom.get()
...
+```
+
+### `atom.swap(state)`
+
+Replace the state with the provided one.
```
Comment (KwanMan) [inline README.md]: "Replace the *entire* state"

File: src/index.js [hunk 4]
```diff
@@ -53,42 +52,33 @@ module.exports = function createAtom (initialState = {}, actions = {}, options =
...
   }
 
-  function createSet (sourceActions) {
+  function createSet (sourceActions, { swap = false } = {}) {
     sourceActions = sourceActions || []
     return function set (update, options = {}) {
       let action = { payload: update }
       let prevState = state
-      state = options.replace ? action.payload : merge(state, action.payload)
+      state = swap ? action.payload : Object.assign({}, state, action.payload)
```
Comment (KwanMan) [inline src/index.js:69]: Do we need to make a copy for `swap`? Could run into something like:

```js
const newState = { foo: 'bar' }
atom.swap(newState)
newState.foo = 'not-bar'
```

**Comment by KwanMan** [comment]
Personally never actually run into a usecase for `swap`/`replace`, but I like the more explicit nature now.

File: src/index.js (hunk 4 above)
Comment (KidkArolis) [reply src/index.js:69]: I'm not too worried about that, you can already mutate state with `atom.get().foo = 'not-bar'` and similar.

Swap is meant to really swap the full reference, useful, when for example, you use Immutable.js where set() with it's Object.assign doesn't make sense to ue. Previously, this was configurable via `options.merge`, but I prefer to remove that option and have these 2 apis instead.

Swap is also useful if you're doing things like removing top level keys from the state or when you're resetting the entire state from local storage, e.g. the `tiny-atom/debug` redux devtools integration does something similar when you click on a historical state entry.

**Pull request merged**